File: imagepipeline-base/src/main/java/com/facebook/imageutils/ImageMetaData.java
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.imageutils;

import android.graphics.ColorSpace;
import android.util.Pair;
import com.facebook.infer.annotation.Nullsafe;
import javax.annotation.Nullable;

/** Wrapper class representing the recovered meta data of an image when decoding. */
@Nullsafe(Nullsafe.Mode.STRICT)
public class ImageMetaData {
  private final @Nullable Pair<Integer, Integer> mDimensions;
  private final @Nullable ColorSpace mColorSpace;

  public ImageMetaData(int width, int height, @Nullable ColorSpace colorSpace) {
    mDimensions = (width == -1 || height == -1) ? null : new Pair<>(width, height);
    mColorSpace = colorSpace;
  }

  public @Nullable Pair<Integer, Integer> getDimensions() {
    return mDimensions;
  }

  public @Nullable ColorSpace getColorSpace() {
    return mColorSpace;
  }
}

File: imagepipeline-base/src/main/java/com/facebook/imageutils/ImageMetaData.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
package com.facebook.imageutils

import android.graphics.ColorSpace

/** Wrapper class representing the recovered meta data of an image when decoding. */
class ImageMetaData(width: Int, height: Int, val colorSpace: ColorSpace?) {

  val dimensions: Pair<Int, Int>? = if (width == -1 || height == -1) null else Pair(width, height)
}
